barretenberg/cpp/src/barretenberg/vm2/simulation/alu.hpp

```diff
@@ -14,7 +14,8 @@ namespace bb::avm2::simulation {
 class AluInterface {
   public:
     virtual ~AluInterface() = default;
-    virtual void add(ContextInterface&, MemoryAddress a_addr, MemoryAddress b_addr, MemoryAddress dst_addr) = 0;
+    // I'd like to return a ValueRefAndTag, but the MemoryValue& doesnt live long enough.
+    virtual FF add(ValueRefAndTag a, ValueRefAndTag b) = 0;
```

Comment:
ValueRefAndTag is probably ok for now, it should probably be a `const&` thought but doesnt matter since we'll change it

Reply:
Question... actually, should we create a new class "ValueAndTag" which holds the FF? At least for now. It's kind of risky having this, because if inside the function you modify memory in any way, your params might get invalidated.

If you are worried about having to construct things when calling, you can make ValueAndTag have an implicit constructor that takes ValueRefAndTag, and the conversion will be automatic.

This is probably safer.

Reply:
Ah duh! the method will not be changing memory so it's likely safe...

Reply:
will leave this for now but updated with ValueRefAndTag with `const&` anyways